File: src/main/java/com/github/kayjamlang/core/expressions/loops/WhileExpression.java
package com.github.kayjamlang.core.expressions.loops;

import com.github.kayjamlang.core.expressions.Expression;
import com.github.kayjamlang.core.opcodes.AccessIdentifier;

public class WhileExpression extends Expression {

    public final Expression condition;
    public final Expression expression;

    public WhileExpression(Expression condition, Expression expression, int line) {
        super(AccessIdentifier.NONE, line);
        this.condition = condition;
        this.expression = expression;
    }
}

File: src/main/scala/com/github/kayjamlang/core/expressions/loops/WhileExpression.scala
package com.github.kayjamlang.core.expressions.loops

import com.github.kayjamlang.core.expressions.Expression
import com.github.kayjamlang.core.opcodes.AccessType

class WhileExpression(val condition: Expression, val expression: Expression, line: Int) extends Expression(AccessType NONE, line)